docs/api/vega-wallet/reference/browser-extension/javascript-api.md

```diff
@@ -0,0 +1,260 @@
+has removed the connection in the meantime.
+
+Once a dApp is finished talking to the wallet it can call 
+`await vega.disconnectWallet()`.
+
+### Detecting the network in use
+
+The Vega browser wallet allows users to have multiple network
+configurations installed at the same time, to allow switching between
+different entry points into the network or different networks entirely 
+(eg. `mainnet`, `fairground`, etc.)
+
+A dApp can detect the network in use by checking the `chainID` in use:
+
+```js
+const { chainID } = await vega.getChainId()
+```
+
+Note that `chainID` is not an indefinate constant but may change with
+network restarts. However it can assumed to constant during a session.
+
+If the dApp is using data from other data-nodes, the `chainID` is a
+great way to compare whether the wallet and dApp are using data from
+the same network.
+
+### Listing available keys
+
+After a successful connection has been established, a dApp can request
+a list of keys approved by the user. The list will include the public key
+and the user given name for the key:
+
+```js
+const keys = await vega.listKeys()
+
+for (const key of keys) {
+    console.log(key.name, key.publicKey)   
+}
+```
+
+The list of keys is useful for dApps to provide contextual information that
+is relevant to the user and a valid public key from the list is required to
+send transactions.
+
+### Sending a transaction
+
+After a successful connection has been establised, a dApp can propose
+transactions for the user to send via their wallet. Note that only
+public keys that appear in the `await vega.listKeys()` list are
+available for the dApp to propose transactions with:
+
+```js
+const res = await vega.sendTransaction({
+  sendingMode: 'TYPE_SYNC', // one of TYPE_ASYNC, TYPE_SYNC, TYPE_COMMIT. See below
+  publicKey: '...', // One of the keys from `listKeys()`
+  transaction: { /* ... */ }
+})
+```
+
+`sendingMode` decides when the Vega node returns a response to the transaction
+submission. `TYPE_ASYNC` will run validation checks by the network in cases 
+where this is possible, but will return fast. `TYPE_SYNC` is the perferred mode.
+This mode will return once the transaction has passed validation checks and 
+entered the mempool.  `MODE_COMMIT` is only available for debug purposes and 
+will require your own node with this mode enabled. This will cause the 
+submission to wait until the transaction has been mined.
+
+`publicKey` must be a key the user has made available to the dApp, otherwise
+the wallet will error with unknown public key.
+
+Transactions in Vega are protocol buffers. The browser wallet provides a JSON
+interface which translates to protobuf fields. As a general rule fields are
+camel-cased, falsy values are not encoded (they map to protobuf default values)
+and optional values can be set to `null` to signify explicit absence.
+
+Below are sample transactions and links to their corresponding `.d.ts` files and
+protobuf comments.
+
+#### Transfer
```

Comment:
@candida-d if we have other documenation for commands should we just link it here rather than duplicate? Guess it depends if the transactions are identical shapes 👍

Reply:
I think they have identical shapes from what I can tell apart from places where protobuf expects (u)int64, in which case web wallet accepts bigints while desktop just takes normal numbers from what I can tell. My intention was for this to be a quick overview, (probably should have been a "getting started" rather than API docs), so I thought it would be good with some tangible examples rather than straight up docs. However I agree, I will cross link to the other docs